classes/Currency.php

```diff
@@ -179,14 +179,28 @@ public function __construct($id = null, $idLang = null, $idShop = null)
             if (null === $idLang) {
                 $idLang = Context::getContext()->language->id;
             }
+            $defaultIdLang = null;
+            if ((is_array($this->symbol) && !isset($this->symbol[$idLang]))
+                || (is_array($this->name) && !isset($this->name[$idLang]))) {
+                $defaultIdLang = Configuration::get('PS_LANG_DEFAULT', null, null, $idShop);
+            }
```

Comment:
What about keeping only this test and replacing `$idLang` if true?

```php
            if ((is_array($this->symbol) && !isset($this->symbol[$idLang]))
                || (is_array($this->name) && !isset($this->name[$idLang]))) {
                $idLang = Configuration::get('PS_LANG_DEFAULT', null, null, $idShop);
            }
```

Reply:
I hesitated, what if `$this->symbol` doesn't have the language, but `$this->name` has it.
Even though I realize it's unlikely.